chronos.py

```diff
@@ -0,0 +1,278 @@
+            })[0])
+        return journals
+
+    def sync_manuscript(self, submission):
+        return self._sync_manuscript(
+            submission,
+            self._client.get_manuscript(submission.publication_id)
+        )
+
+    def get_journals(self):
+        return ChronosJournal.objects.all()
+
+    def submit_manuscript(self, journal, preprint, submitter):
+        if ChronosSubmission.objects.filter(journal=journal, preprint=preprint).exists():
+            raise ValueError('{!r} already has an existing submission to {!r}. Use {!r}.update_manuscript'.format(preprint, journal, self))
+
+        body = ChronosSerializer.serialize_manuscript(journal.journal_id, preprint)
+        body['USER'] = ChronosSerializer.serialize_user(submitter)
+
+        response = self._client.submit_manuscript(body)
+
+        with transaction.atomic():
+            submission = ChronosSubmission.objects.create(
+                journal=journal,
+                preprint=preprint,
+                submitter=submitter,
+                raw_response=response,
+                # Things parsed out of response
+                publication_id=response['PUBLICATION_ID'],
+                status=response['STATUS_CODE'],
+                submission_url=response['CHRONOS_SUBMISSION_URL'],
+            )
+
+            submitter.chronos_user_id = response['USER']['CHRONOS_USER_ID']
+            submitter.save()
+
+            for contrib, author in zip(preprint.node.contributor_set.filter(visible=True).select_related('user'), response['AUTHORS']):
+                assert author['PARTNER_USER_ID'] == contrib.user._id
+                contrib.user.chronos_user_id = author['CHRONOS_USER_ID']
+                contrib.user.save()
+
+        return submission
+
+    def update_manuscript(self, submission):
+        body = ChronosSerializer.serialize_manuscript(submission.journal.journal_id, submission.preprint, status=submission.status)
+        body['USER'] = ChronosSerializer.serialize_user(submission.submitter)
+        body['PUBLICATION_ID'] = submission.publication_id
+
+        return self._sync_manuscript(
+            submission,
+            self._client.update_manuscript(body),
+        )
+
+    def _sync_manuscript(self, submission, response):
+        with transaction.atomic():
+            # TODO pick of any interesting fields
+            submission.status = response['STATUS_CODE']
+            # Not present when fetching from the API
+            if response['CHRONOS_SUBMISSION_URL']:
+                submission.submission_url = response['CHRONOS_SUBMISSION_URL']
+            submission.save()
+        return submission
+
+
+class ChronosRestClient(object):
+
+    def __init__(self, username, password, api_key, host='http://sandbox.api.chronos-oa.com'):
+        self._auth_key = None
+        self._session = requests.Session()
+
+        self._api_key = api_key
+        self._host = host
+        self._password = password
```

Comment:
Since this is expected to only ever be the "OSF system" user, consider adding settings for `CHRONOS_PASSWORD`, etc